src/algebra/big_operators/order.lean

```diff
@@ -308,3 +309,50 @@ open opposite
 (@unop_add_hom β _).map_sum _ _
 
 end with_top
+
+namespace finset
+
+/--
+The strong pigeonhole principle for finitely many pigeons and pigeonholes.
```

Comment:
Once we agree on the right statement, let's state it completely in the doc-string.

Reply:
Stated.